### PR TITLE
assisted-service: Extend dual-stack jobs

### PR DESCRIPTION
**Pull request by mkowalski**
This PR extends dual-stack jobs in the following ways

* dual-stack SNO installing 4.8 is running for every PR as optional
* all dual-stack jobs are now available as 4.8 and 4.9

This PR starts to introduce a shorter naming convention for Assisted
Installer CI jobs. It makes the following changes for the dual-stack
jobs

* `metal-assisted` becomes `ai`
* `dualstack` becomes `v4v6`
* `multinode-spoke` becomes `3node`

/cc @lranjbar

### CODE REVIEW

**Comment by mkowalski** [comment]
/retest-required

Packet setup failed.

**Comment by mkowalski** [comment]
/test pj-rehearse

Packet setup failed.

**Review by flaper87 — COMMENTED** [review]
My only bikeshed idea is `3node` vs `compact`. I have a slight preference for the latter as we could name this part like:

- sno
- compact
- compact+$NUMBER_OF_WORKERS
- full (3 masters, 3 workers)

On one hand I like that 3node explicitly says how many nodes there are. I don't feel too strong either way/

**Comment by mkowalski** [comment]
>My only bikeshed idea is 3node vs compact. I have a slight preference for the latter as we could name this part like:

As discussed, I introduce the following convention

* `1m` - 1 master, effectively a SNO cluster
* `3m` - 3 masters, a clusters where master and workers are the same node

Following this, in the future we could have

* `3m3w` that would be a full OCP cluster with 3 masters and 3 separate workers

**Comment by YuviGold** [comment]
@mkowalski 
1. I like the shortenings. But I'd suggest to change from `v4v6` to `ipv4v6` since it's not clear what versions are being talked about :) 
2. `1m` usually stands for `1 minute`... what's wrong with the very known `sno`? :)
3. Why do we need 2 presubmits for every OCP release? Usually all of our jobs don't define explicitly OCP version that should be independnt. presubmits are running against the stable version (currently 4.8) and periodics run against the candidate version (currently 4.9) . This way presubmits aren't blocked and we still monitor what happens with our candidates.
I don't think that we need to define a job for every release image as it will be become hard to follow and our logic shouldn't depend on a version.

**Comment by mkowalski** [comment]
>1m usually stands for 1 minute... what's wrong with the very known sno? :)

`sno` instead of `1m` is good, I'll change that. How should we handle the "3 master cluster" ? I did not manage to invent anything better than `3m` and `3masters` will already start to create long names (the problem we initially had to solve)

>Usually all of our jobs don't define explicitly OCP version that should be independnt. presubmits are running against the stable version (currently 4.8) and periodics run against the candidate version (currently 4.9)

Right, I was thinking about this, but the main argument here was that dual-stack is still something that is in the process of developing in the upstream OCP and as for today I cannot say if there will be always a feature-parity between 4.8, 4.9, 4.10 and so on (e.g. regards the number and order of networks). For this reason I wanted to have CI running against multiple of them, because sometimes it may happen that we implicitly assume some feature to be there, but it won't be supported by _some_ OCP release.

The logic I tried to follow was

* 1 job running for every PR in AI as a sanity check
* daily CI running against some of the combinations
* every possible combination available manually under `/test` (I still need to add a job that would run against OCP 4.10, i.e. master/nightly)

**Comment by flaper87** [comment]
> `sno` instead of `1m` is good, I'll change that. How should we handle the "3 master cluster" ? I did not manage to invent anything better than `3m` and `3masters` will already start to create long names (the problem we initially had to solve)

- Maybe `3M` (uppercase) may help with the ambiguity? 
- Anothoer option is to always be explicit about workers `3m0w` `3m1w` etc

Otherwise, let's just go with `3masters` and avoid bikeshedding for too long on this. I feel comfortable saying this since I feel like this bikeshed is all my fault (although I think it's important in this case) :)

**Comment by mkowalski** [comment]
`rehearse-22315-pull-ci-openshift-assisted-service-release-ocm-2.4-e2e-ai-operator-ztp-ipv4v6-3masters-ocp-48` failed for an unrelated issue

```
error: unable to parse image registry.build01.ci.openshift.org/ci-op-fvjtp55i/stable@sha256:bc217dd0073df5e396866b52e4d7c05a6f74ea2291fd6f1b4bb78c94cabd009f: cannot retrieve image configuration for manifest sha256:bc217dd0073df5e396866b52e4d7c05a6f74ea2291fd6f1b4bb78c94cabd009f: received unexpected HTTP status: 500 Internal Server Error
```
